fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/NotValue.java

```diff
@@ -99,7 +105,7 @@ public Optional<QueryPredicate> toQueryPredicate(@Nullable final TypeRepository
     @Nonnull
     @Override
     public Iterable<? extends Value> getChildren() {
-        return ImmutableList.of(child);
+        return childrenSupplier.get();
```

Comment:
All `Value`s extend from `AbstractValue`. Can you put all suppliers in there, define an abstract `computeChildren()` and have that memoized.

Reply:
I think I wanted to avoid using a `Supplier` where it is superfluous to do so. But what you suggest is cleaner, so I refactored the code accordingly.